src/detect-engine-siggroup.c

```diff
@@ -801,58 +797,35 @@ static int SigGroupHeadTest03(void)
  */
 static int SigGroupHeadTest06(void)
 {
-    int result = 1;
     SigGroupHead *sh = NULL;
+
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    Signature *prev_sig = NULL;
+    FAIL_IF_NULL(de_ctx);
 
-    if (de_ctx == NULL)
-        return 0;
+    Signature *s = DetectEngineAppendSig(de_ctx, "alert tcp any any -> any any "
```

Comment:
What's the reasoning for converting the test to `DetectEngineAppendSig` from `SigInit`?